conjure-api/src/main/conjure/conjure-api.yml

```diff
@@ -8,6 +8,7 @@ types:
           errors: list<ErrorDefinition>
           types: list<TypeDefinition>
           services: list<ServiceDefinition>
+          extensions: map<string, any>
```

Comment:
I kinda want to put a floating type in the IR which is a ProductDependency, that people can json convertValue one of these objects into... otherwise we'll be re-defining the same thing so many times

Reply:
I'd prefer to not add the type to Conjure since in practice you'd want language specific bindings for `SlsVersion` and `SlsVersionMatcher`. It also re-enforces that things within the extension are outside of the spec